plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/metastore/TestDeltaLakeMetastoreAccessOperations.java

```diff
@@ -254,10 +294,286 @@ public void testShowTables()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_cache_metastore", "(col int)")) {
+            assertMetastoreInvocations("ALTER TABLE " + table.getName() + " EXECUTE optimize",
+                    ImmutableMultiset.<MetastoreMethod>builder()
+                            .add(GET_TABLE)
+                            .add(REPLACE_TABLE)
+                            .build());
+        }
+    }
+
+    @Test
+    public void testCacheMetastoreVacuum()
+    {
+        Session sessionWithShortRetentionUnlocked = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "vacuum_min_retention", "0s")
+                .build();
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_cache_metastore", "AS SELECT 1 a")) {
+            assertUpdate("UPDATE " + table.getName() + " SET a = 2", 1);
+            assertMetastoreInvocations(sessionWithShortRetentionUnlocked,
+                    "CALL system.vacuum(schema_name => CURRENT_SCHEMA, table_name => '" + table.getName() + "', retention => '0s')",
+                    ImmutableMultiset.<MetastoreMethod>builder()
+                            .add(GET_TABLE)
+                            .build());
+        }
+    }
+
+    @Test
+    public void testCacheMetastoreRegisterTable()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_cache_metastore", "(col int) COMMENT 'test comment'")) {
+            assertUpdate("INSERT INTO " + table.getName() + " VALUES 1", 1);
+            String tableLocation = metastore.getTable(TPCH_SCHEMA, table.getName()).orElseThrow().getStorage().getLocation();
+            metastore.dropTable(TPCH_SCHEMA, table.getName(), false);
+
+            assertMetastoreInvocations("CALL system.register_table('%s', '%s', '%s')".formatted(TPCH_SCHEMA, table.getName(), tableLocation),
+                    ImmutableMultiset.<MetastoreMethod>builder()
+                            .add(GET_DATABASE)
+                            .add(CREATE_TABLE)
+                            .build());
+        }
+    }
+
+    @Test
+    public void testCacheMetastoreDataManipulation()
+    {
+        String schemaString = "{\"type\":\"struct\",\"fields\":[{\"name\":\"col\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}]}";
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_cache_metastore", "(col int)")) {
+            assertThat(metastore.getTable(TPCH_SCHEMA, table.getName()).orElseThrow().getParameters())
+                    .contains(entry("trino_last_transaction_version", "0"), entry("trino_metadata_schema_string", schemaString));
+
+            assertMetastoreInvocations("INSERT INTO " + table.getName() + " VALUES 1",
+                    ImmutableMultiset.<MetastoreMethod>builder()
+                            .add(GET_TABLE)
+                            .add(REPLACE_TABLE)
+                            .build());
```

Comment:
And without cache, we do only GET_TABLE, right?

would it be possible to have "do cache metadata" as a boolean variable in the test
 so that the expected can be expressed as a conditional expression based on "do cache metadata"? this would make the test tell the story more directly